code/game/objects/structures/beacon.dm

```diff
@@ -0,0 +1,45 @@
+/obj/structure/strangebeacon
+	icon = 'icons/obj/machines/excelsior/objects.dmi'
+	icon_state = "strange_beacon"
+	var/nosignal = 0
+
+/obj/structure/strangebeacon/attack_hand(mob/living/user as mob)
+	if(nosignal == 0)
+		playsound(loc, 'sound/machines/twobeep.ogg', 50, 1)	//Plays a beep
+		nosignal = 1
+
+/obj/structure/strangebeacon/bots //who you gonna call
```

Comment:
FYI you actually don't need to define type explicitly like this, if you just declare a proc (e.g. `/obj/structure/strangebeacon/bots/attack_hand`) it'll make a subtype properly.